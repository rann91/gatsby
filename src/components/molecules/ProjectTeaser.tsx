/* @jsx jsx */
import { Fragment } from 'react'
import { jsx, AspectRatio, Heading, Text } from 'theme-ui'
import { Strapi_Project } from '../../typings/graphql'
import { Link } from 'gatsby'
import Image, { FluidObject } from 'gatsby-image'

const ProjectTeaser = ({ title, slug, category, image }: Strapi_Project) => {
  const url = `/work/${slug}`

  return (
    <Fragment>
      <Link to={url}>
        <AspectRatio ratio={16 / 9} bg="black">
          {image && (
            <Image
              fluid={image?.file?.childImageSharp?.fluid as FluidObject}
              alt={image?.alternativeText || undefined}
            />
          )}
        </AspectRatio>
      </Link>
      {title && (
        <Heading
          as="h3"
          variant="h3"
          pt={[5, null, null, 6]}
          sx={{ textAlign: 'center' }}>
          <Link to={url} sx={{ variant: 'text.headingLink' }}>
            {title}
          </Link>
        </Heading>
      )}
      {category && (
        <Text as="p" variant="p" pt={2} sx={{ textAlign: 'center' }}>
          {category}
        </Text>
      )}
    </Fragment>
  )
}

export default ProjectTeaser
