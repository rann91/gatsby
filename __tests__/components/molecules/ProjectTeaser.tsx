import { render } from '@testing-library/react'
import React from 'react'
import ProjectTeaser from '../../../src/components/molecules/ProjectTeaser'
import project from '../../fixtures/project.json'

describe('ProjectTeaser', () => {
  it('renders correctly', () => {
    const { container } = render(<ProjectTeaser {...project} />)

    expect(container).toMatchSnapshot()
  })
})
