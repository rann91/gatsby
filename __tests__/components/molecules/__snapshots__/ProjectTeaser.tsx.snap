// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTeaser renders correctly 1`] = `
<div>
  <a
    href="/work/undefined"
  >
    <div
      class="css-i9l3dq"
    >
      <div
        class="css-r01jhj"
      />
      <div
        class="css-1sr1905"
      />
    </div>
  </a>
</div>
`;
